tests/unit/config_test.py

```diff
@@ -407,7 +422,7 @@ def test_nonexistent_path(self):
         options = {'build': 'nonexistent.path'}
         self.assertRaises(
             config.ConfigurationError,
-            lambda: config.make_service_dict('foo', options, 'tests/fixtures/build-path'),
+            lambda: config.validate_paths(config.make_service_dict('foo', options, 'tests/fixtures/build-path')),
```

Comment:
Could this lambda call `from_dictionary` instead of `valid_paths(make_service_dict)`? I'm worried about reaching too far into the internals - it makes the tests more stable, but could hide regressions if the code is restructured.